### PR TITLE
Mat 2150 threshold calculation

### PR DESCRIPTION
**Pull request by patrickherring-TRI**
This PR adds a function to calculate the cycle index and throughput at which the run drops below a threshold for the remaining fractional capacity.
- Function to calculate the values for the threshold targets for early prediction
- Ability to extrapolate for cells that have not yet reached the threshold
Screen Shot 2020-12-17 at 12 35 06 AM

### CODE REVIEW

File: beep/dataset.py [hunk 1]
```diff
@@ -339,6 +340,104 @@ def serialize(self):
         return self.dataset_dir
 
 
+def get_threshold_targets(dataset_diagnostic_properties,
+                          cycle_type_target="rpt_1C",
```
Comment (chirranjeevigopal-TRI) [inline beep/dataset.py:344]: Just to be consistent with the naming (with other modules), can we name this `cycle_type`?

File: beep/dataset.py [hunk 2]
```diff
@@ -339,6 +340,104 @@ def serialize(self):
...
+                          basis="discharge_energy",
```
Comment (chirranjeevigopal-TRI) [inline beep/dataset.py:345]: Again, for the sake of consistency with other modules and method, call `basis` ->`metric`?

File: beep/dataset.py [hunk 3]
```diff
@@ -339,6 +339,87 @@ def serialize(self):
...
+                          threshold=0.80,
+                          filter_kinks=None,
+                          extrapolate_threshold=True):
```
Comment (BruisVanVlijmen2020) [inline beep/dataset.py:348]: If not `None`, what could `filter_kinks` be for example?

Comment (patrickherring-TRI) [reply beep/dataset.py:348]: I found for some of the cells a value of 0.04 seemed to be a good choice. Obviously depends a lot on what conditions you are trying to filter out.

File: beep/dataset.py (hunk 2 above)
Comment (patrickherring-TRI) [reply beep/dataset.py]: yup, done!

File: beep/dataset.py (hunk 1 above)
Comment (patrickherring-TRI) [reply beep/dataset.py]: done!

**Review by chirranjeevigopal-TRI — APPROVED** [review]
lgtm!

**Comment by BruisVanVlijmen2020** [comment]
looks good to me too

**Pull request merged**